agent/vendor/github.com/aws/amazon-ecs-agent/ecs-agent/acs/session/heartbeat_responder.go

```diff
@@ -11,40 +11,64 @@
 // express or implied. See the License for the specific language governing
 // permissions and limitations under the License.
 
-package handler
+package session
 
 import (
 	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
 	"github.com/aws/amazon-ecs-agent/ecs-agent/doctor"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/logger/field"
 	"github.com/aws/amazon-ecs-agent/ecs-agent/wsclient"
 	"github.com/aws/aws-sdk-go/aws"
-	"github.com/cihub/seelog"
 )
 
-func HeartbeatHandlerFunc(acsClient wsclient.ClientServer, doctor *doctor.Doctor) func(message *ecsacs.HeartbeatMessage) {
-	return func(message *ecsacs.HeartbeatMessage) {
-		handleSingleHeartbeatMessage(acsClient, doctor, message)
+// heartbeatResponder implements the wsclient.RequestResponder interface for responding
+// to ecsacs.HeartbeatMessage type.
+type heartbeatResponder struct {
+	doctor  *doctor.Doctor
+	respond wsclient.RespondFunc
+}
+
+// NewHeartbeatResponder returns an instance of the heartbeatResponder struct.
+func NewHeartbeatResponder(doctor *doctor.Doctor) *heartbeatResponder {
```

Comment:
I'm wondering if this should be returning `wsclient.RequestResponder` instead. One benefit I see for that is the type protection that this function would provide.

Reply:
Yeah, we can have it return `wsclient.RequestResponder` instead. We will need to update test(s) to do a type assertion that the returned responder is of type `*heartbeatResponder` so that we can still call the `processHeartbeatMessage` function there. I'll make these updates accordingly.